examples/lfric/full_example_netcdf/testkern_w0_mod.f90

```diff
@@ -33,31 +28,39 @@
 ! OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 ! OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 ! -----------------------------------------------------------------------------
-! Modified R. W. Ford, STFC Daresbury Laboratory
-!          I. Kavcic, Met Office
+! Author R. Ford STFC Daresbury Lab
 
-!> @brief Define enumerator variables that describe the different types of
-!!        function space continuity.
+module testkern_w0_mod
+  use argument_mod
+  use kernel_mod
+  use fs_continuity_mod, only: W0
 
-module fs_continuity_mod
+  use constants_mod
+  type, extends(kernel_type) :: testkern_w0_type
+     type(arg_type), dimension(2) :: meta_args =  &
+          (/ arg_type(gh_field,gh_inc, w0),       &
+             arg_type(gh_field,gh_read,w0)        &
+           /)
+     integer :: iterates_over = cells
+   contains
+     procedure, nopass :: code => testkern_w0_code
+  end type testkern_w0_type
+contains
 
-  use constants_mod, only: i_native
+  subroutine testkern_w0_code(nlayers, fld1, fld2, ndf_w0, undf_w0, map_w0)
+    integer :: nlayers
+    real(kind=r_def), dimension(:), intent(inout) :: fld1
+    real(kind=r_def), dimension(:), intent(in)    :: fld2
+    integer :: ndf_w0, undf_w0
+    integer, dimension(:) :: map_w0
 
-  implicit none
+    integer :: i, k
+    do i=1, ndf_w0
+      do k=0, nlayers-1
```

Comment:
`k` loop over layers is the outermost one in LFRic kernels.

Reply:
Done.